### PR TITLE
Clean tests and other authentication mechanisms

### PR DESCRIPTION
**Pull request by naveentatikonda**
Signed-off-by: Naveen Tatikonda <navtat@amazon.com>

### Description
This PR will clean up all the authentication mechanisms not supported by OpenSearch and test specs.

### Issues Resolved
https://github.com/opensearch-project/logstash-input-opensearch/issues/8

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/opensearch_helper.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module OpenSearchHelper
   def self.get_host_port
     if ENV["INTEGRATION"] == "true" || ENV["SECURE_INTEGRATION"] == "true"
-      "opensearch:9200"
+      "integration:9200"
```
Comment (vamshin) [inline spec/opensearch_helper.rb:4]: why is this changed?

Comment (naveentatikonda) [reply spec/opensearch_helper.rb:4]: In the docker-compose file in build scripts, we have defined service name as "integration". So, the service url also needs to be updated. I will be pushing the build scripts in my next PR

**Review by Shivamdhar — APPROVED** [review]

**Comment by VijayanB** [comment]
Why CI is not running? Can you enable CI first before making any changes? Thanks

**Comment by naveentatikonda** [comment]
These changes are committed as part of some other PRs, so closing it
